src/shared/autocomplete/autocomplete.js

```diff
@@ -266,6 +266,9 @@ export class AutoComplete {
         }
 
         const list = typeof this._list === "function" ? await this._list() : this._list;
+        if (typeof (list) === 'undefined') {
+            return 0;
+        }
```

Comment:
@jcbrand, so given that the [jids](https://github.com/conversejs/converse.js/pull/2881/files#diff-10ea71a0f5e70bf68e685adee44d6718e8e6f408175f67a59be213eaf7b5f7deR131) is undefined, we were accessing its length, which threw an error, so I simply handled it via returning a 0.